src/controllers/satisfactioncontroller.js

```diff
@@ -0,0 +1,50 @@
+import Controller from './core/controller';
+import CustomerController from './customercontroller';
+import ScoreController from "./scorecontroller";
+
+export default class SatisfactionController extends Controller
+
+{
+    /**
+     * Updates player satisfaction after every completed order, as well as the score
+     * @param customer
+     */
+    updatePlayerSatisfaction(customer)
+    {
+        let config = GAME.model.config;
+
+        const scoreController = new ScoreController();
+
+        if (customer.satisfaction > 100)
+            customer.satisfaction = 100;
+
+        config.playerSatisfaction = ((config.playerSatisfaction * config.completedOrders) + (customer.satisfaction))
+                                    / (config.completedOrders + 1);
+
+        $("#satisfaction").html(GAME.model.config.playerSatisfaction.toFixed(0));
+
+        scoreController.updateScore();
+    }
+
+    /**
+     * Updates satisfaction of customers each time a new day occurs, and adds appropriate emojis.
+     */
+    updateCustomerSatisfaction()
+    {
+        GAME.model.customers.forEach(function (customer) {
+            customer.satisfaction -= 10;
+
+            const customerController = new CustomerController();

```

Comment:
Only used in the if statement, so init it there (block scoping is better than function-level globals).

Reply:
Yeah, I still have the tendency to init things all at the same place/beginning.